openedx/features/course_experience/static/course_experience/js/Enrollment.js

```diff
@@ -0,0 +1,42 @@
+
+/*
+ * Course Enrollment on the Course Home page
+ */
+export class CourseEnrollment {  // eslint-disable-line import/prefer-default-export
+  /**
+   * Redirect to a URL.  Mainly useful for mocking out in tests.
+   * @param  {string} url The URL to redirect to.
+   */
+  static redirect(url) {
+    window.location.href = url;
+  }
+
+  static refresh() {
+    window.location.reload(false);
+  }
+
+  static createEnrollment(courseId) {
+    const data = JSON.stringify({
+      course_details: { course_id: courseId },
+    });
+    const enrollmentAPI = '/api/enrollment/v1/enrollment';
+    const trackSelection = '/course_modes/choose/';
+
+    return () =>
+      $.ajax(
+        {
+          type: 'POST',
+          url: enrollmentAPI,
+          data,
+          contentType: 'application/json',
+        }).done(() => {
+          CourseEnrollment.refresh();
+        }).fail(() => {
+          CourseEnrollment.redirect(trackSelection + courseId);
```

Comment:
Separately, I thought we were going to have this page handle errors?  Seems odd to redirect to track selection, but maybe that is what we need to do for now to keep this simple?  Depending on the decision, we may want to drop an appropriate comment.